src/app.py

```diff
@@ -186,27 +86,98 @@ def schema(xsdfile):
+
+
+@app.route('/rgdowntime/xml')
+def rgdowntime_xml():
+    return _get_xml_or_fail(global_data.get_topology().get_downtimes, request.args)
+
+
+@app.route('/downtime_resource_select')
+def downtime_resource_select():
+    path = "downtime_resource_select"
+    template = f"{path}_form.html"
+
+    form = DowntimeResourceSelectForm(request.form)
+    topo = global_data.get_topology()
+    form.facility.choices = _make_choices(topo.resource_names_by_facility.keys())
+
+    facility = request.args.get("facility", "")
+    if facility:
+        resource_names = topo.resource_names_by_facility.get(facility)
+        if not resource_names:
+            return make_response((f"""
+Missing or invalid facility. <a href="/{path}">Select a facility.</a>
+""", 400))
+        form.facility.data = facility
+        form.resource.choices = _make_choices(resource_names)
+        return render_template(template, form=form, facility=facility)
+    return render_template(template, form=form)
+
+
+@app.route("/generate_downtime", methods=["GET", "POST"])
+def generate_downtime():
+    path = "generate_downtime"
+    template = f"{path}_form.html"
+
+    form = GenerateDowntimeForm(request.form)
+    topo = global_data.get_topology()
+
+    facility = request.args.get("facility")
+    if not facility or facility not in topo.resource_names_by_facility:
+        return make_response((f"""
+Missing or invalid facility. <a href="/{path}">Select a facility.</a>
+""", 400))
+    resource = request.args.get("resource")
+    resource_names = topo.resource_names_by_facility[facility]
+    if not resource or not resource_names or resource not in resource_names:
+        return make_response((f"""
+Missing or invalid resource in facility {flask.escape(facility)}.
+<a href="/{path}?facility={urllib.parse.quote(facility)}">Select a resource</a>
+or <a href="/{path}">select another facility.</a>
+""", 400))
+
+    form.resource.data = resource
+
     try:
-        filters = get_filters_from_args(request.args)
-    except InvalidArgumentsError as e:
-        return Response("Invalid arguments: " + str(e), status=400)
+        form.services.choices = _make_choices(topo.service_names_by_resource[resource])
+    except (KeyError, IndexError):  # shouldn't happen but deal with anyway
+        return make_response((f"""
+Missing or invalid services in resource {flask.escape(resource)}.
+<a href="/{path}?facility={urllib.parse.quote(facility)}">Select another resource</a>
+or <a href="/{path}">select another facility.</a>
+""", 400))
+
+    if not form.validate_on_submit():
+        return render_template(template, form=form, resource=resource, facility=facility)
 
-    authorized = _get_authorized()
-    vos_xml = to_xml_bytes(global_data.get_vos_data().get_tree(authorized, filters))
-    return Response(vos_xml, mimetype='text/xml')
+    filename = "topology/" + topo.downtime_path_by_resource[resource]
```

Comment:
Can we make this a little smarter? 

- If the file exists, let's link them directly to the edit form, e.g. https://github.com/opensciencegrid/topology/edit/master/topology/University%20of%20Wisconsin/GLOW/GLOW_downtime.yaml.
- If the file doesn't exist:
    1. Link them to the relevant site dir, e.g. https://github.com/opensciencegrid/topology/tree/master/topology/University%20of%20Wisconsin/GLOW
   1. Tell them to click the "Create new file" and give them the appropriate downtime name